pkg/resources/resource_user.go

```diff
@@ -0,0 +1,298 @@
+	d.Set("verified", user.Verified)
+	d.Set("metadata", user.Metadata)
+
+	return nil
+}
+
+// TODO: Add userUpdate function to change user roles
+
+// userDelete is the Terraform resource delete function for a Frontegg user.
+func userDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerMeta, ok := utils.GetProviderMeta(meta)
+	if !ok {
+		return diag.Errorf("expected meta to be a struct containing DB and Frontegg, got %T", meta)
+	}
+	client := providerMeta.Frontegg
+
+	userID := d.Id()
+
+	// Send the request to the Frontegg API to delete the user.
+	req, err := http.NewRequestWithContext(ctx, "DELETE", fmt.Sprintf("%s/identity/resources/users/v1/%s", client.Endpoint, userID), nil)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating request to delete user: %s", err))
+	}
+	req.Header.Add("Authorization", "Bearer "+client.Token)
+
+	// Perform the request
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error sending request to delete user: %s", err))
+	}
+	defer resp.Body.Close()
+
+	// Check for a successful response
+	if resp.StatusCode != http.StatusOK {
+		return diag.FromErr(fmt.Errorf("error deleting user: status %d", resp.StatusCode))
+	}
+
+	// Remove the user from the Terraform state
+	d.SetId("")
+	return nil
+}
+
+// convertToStringSlice is a helper function to convert an interface slice to a string slice.
+func convertToStringSlice(input []interface{}) []string {
+	result := make([]string, len(input))
+	for i, v := range input {
+		result[i] = v.(string)
+	}
+	return result
+}
+
+// listRoles fetches roles from the Frontegg API and returns a map of role names to their IDs.
+func listRoles(ctx context.Context, client *clients.FronteggClient) (map[string]string, error) {
+	rolesURL := fmt.Sprintf("%s/identity/resources/roles/v2", client.Endpoint)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", rolesURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %v", err)
+	}
+	req.Header.Add("Authorization", "Bearer "+client.Token)
+
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error executing request: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("error fetching roles, status code: %d", resp.StatusCode)
+	}
+
+	// Read and reset the response body
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading response body: %v", err)
+	}
+	resp.Body = io.NopCloser(bytes.NewBuffer(responseBody))
+
+	// Decode the JSON response
+	var rolesResponse FronteggRolesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&rolesResponse); err != nil {
+		return nil, fmt.Errorf("error decoding response: %v", err)
+	}
+
+	// Create a map of role names to their IDs
+	roleMap := make(map[string]string)
+	for _, role := range rolesResponse.Items {
```

Comment:
Does this work with multiple roles? Won't the map just override the latest `Admin` and `Member` keys with the latest role id with those roles?

Reply:
The idea behind this is to map the shorthand syntax eg. `Admin` and `Member` to the actual response from the API, eg:

```
Roles response: {[{11111111-aaaaa-11111-aaaaa-1111111 Organization Admin} {22222222-bbbbb-11111-aaaa-2222222 Organization Member}] {2 1}}
```

That way in the schema we could require `Admin` and/or `Member` instead of asking for the `Organization` prefix as well.

Probably there is a better way of handing this!